python/nav/mibs/hpicf_powersupply_mib.py

```diff
@@ -13,73 +13,100 @@
 # details.  You should have received a copy of the GNU General Public License
 # along with NAV. If not, see <http://www.gnu.org/licenses/>.
 #
-""" Implements a MibRetriever for the POWERSUPPLY-MIB, as well as helper
-classes.
-
-POWERSUPPLY-MIB is a mib that can be downloaded from HP's support pages.
-"""
+"""Implements a MibRetriever for Hewlett Packard's POWERSUPPLY-MIB."""
+from operator import attrgetter
 
 from twisted.internet import defer
 
+from django.utils import six
+
 from nav.mibs import reduce_index
+from nav.mibs.entity_mib import EntityMib
 from nav.smidumps import get_mib
 from nav.mibs import mibretriever
 
 from nav.models.manage import PowerSupplyOrFan as PSU
 
 
+PSU_STATUS_MAP = {
+    "psPowered": PSU.STATE_UP,
+    "psNotPlugged": PSU.STATE_DOWN,
+    "psFailed": PSU.STATE_DOWN,
+    "psPermFailure": PSU.STATE_DOWN,
+    "psMax": PSU.STATE_WARNING,
+}
+
+
 class HpIcfPowerSupplyMib(mibretriever.MibRetriever):
-    """ A class for collecting powersupply states from HP netboxes."""
-    mib = get_mib('POWERSUPPLY-MIB')
+    """A MibRetriever for collecting power supply states from HP netboxes."""
+
+    mib = get_mib("POWERSUPPLY-MIB")
 
     def __init__(self, agent_proxy):
-        """ Constructor, anything more to say...?"""
         super(HpIcfPowerSupplyMib, self).__init__(agent_proxy)
+        self.entity_mib = EntityMib(agent_proxy)
         self.psu_status_table = None
 
     @defer.inlineCallbacks
     def _get_psu_status_table(self):
-        """Return the powersupply-status table from this netbox."""
-        df = self.retrieve_table('hpicfPsTable')
-        df.addCallback(self.translate_result)
-        df.addCallback(reduce_index)
-        psu_table = yield df
-        self._logger.debug('psu_table: %s' % psu_table)
+        """Returns the power supply status table from this netbox."""
```

Comment:
The old way is faster to debug since you can introspect on "df".

Reply:
Agreed. Fixed.